main.go

```diff
@@ -291,17 +316,19 @@ func newPatchRunner(fset *token.FileSet, patches []*engine.Program) *patchRunner
 	}
 }
 
-func (r *patchRunner) Apply(filename string, f *ast.File) (*ast.File, bool) {
+func (r *patchRunner) Apply(filename string, f *ast.File) (*ast.File, string, bool) {
 	snap := astdiff.Before(f, ast.NewCommentMap(r.fset, f, f.Comments))
 
 	matched := false
+	comments := ""
 	for _, prog := range r.patches {
 		for _, c := range prog.Changes {
 			d, ok := c.Match(f)
 			if !ok {
 				// This patch didn't modify the file. Try the next one.
 				continue
 			}
+			comments += strings.Join(c.Comments, "\n")
```

Comment:
I suggest we don't change representation here and leave it to the representation layer.